tests/e2e/conftest.py

```diff
@@ -1,17 +1,55 @@
 """Additional arguments for pytest."""
 
+import pytest
+from pytest import TestReport
+
+import tests.e2e.test_api as test_api
+
+makereport_called = False
+
+
+def pytest_runtest_makereport(item, call) -> TestReport:
+    """Generate a synthetic test report.
+
+    If OLS did not come up in time generate a synethic test entry,
+    generate a normal test report entry otherwise.
+    """
+    global makereport_called
+    # The first time we try to generate a test report, check if OLS timed out on startup
+    # if so, generate a synthetic test report for the wait_for_ols timeout.
+    if not test_api.OLS_READY and not makereport_called:
+        makereport_called = True
+        return TestReport(
+            "test_wait_for_ols",
+            ["", 0, ""],
+            None,
+            "failed",
+            "wait for OLS to startup before running tests",
+            "call",
+            [],
+        )
+    # The second time we are called to generate a report, assuming OLS timed out,
+    # exit pytest so we don't try to run any more tests (they will fail anyway since
+    # OLS didn't come up)
+    # There is no clean way to return the synthetic test report above *and* exit pytest
+    # in a single invocation
+    if not test_api.OLS_READY:
+        pytest.exit("OLS did not become ready!", 1)
+    # If OLS did come up cleanly during setup, then just generate normal test reports for all tests
+    return TestReport.from_item_and_call(item, call)
+
 
 def pytest_addoption(parser):
     """Argument parser for pytest."""
     parser.addoption(
         "--eval_provider",
-        default="openai",
+        default="watsonx",
```

Comment:
not sure where this change came from but i don't think it should be part of this PR